### PR TITLE
feat: select gas currency ui

### PR DESCRIPTION
**Pull request by nuanyang233**
## Description

Closes #

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/shared/src/UI/components/SelectGasSettingsToolbar/index.tsx [hunk 1]
```diff
@@ -243,7 +247,7 @@ export function SelectGasSettingsToolbarUI({
     }, [gasLimit, gasOption, currencyRatio, currentGasCurrency, nativeToken])
 
     const gasFeeUSD = useMemo(() => {
-        if (!gasFee) return '0'
+        if (!gasFee || gasFee.isZero()) return '0'
```
Comment (guanbinrui) [inline packages/shared/src/UI/components/SelectGasSettingsToolbar/index.tsx:250]: isZero may able to check both emptiness and zero.

File: packages/plugins/SmartPay/src/SNSAdaptor/components/SmartPayContent.tsx [hunk 2]
```diff
@@ -308,21 +306,20 @@ export const SmartPayContent = memo(() => {
...
-
-        setCurrent((prev) => {
-            if (!prev) return first(contractAccounts)
-            return prev
-        })
-    }, [contractAccounts])
+        if (account) return
+        const contractAccount = first(contractAccounts)
+        if (!contractAccount) return
+        setAccount(contractAccount.address)
```
Comment (guanbinrui) [inline packages/plugins/SmartPay/src/SNSAdaptor/components/SmartPayContent.tsx]: Avoid setState in useEffect.

File: packages/web3-providers/src/types/AbstractAccount.ts [hunk 3]
```diff
@@ -72,6 +72,7 @@ export namespace AbstractAccountAPI {
             owner: string,
             transaction: Transaction,
             signer: Signer<ECKeyIdentifier> | Signer<string>,
+            gasCurrency?: string,
```
Comment (guanbinrui) [inline packages/web3-providers/src/types/AbstractAccount.ts:75]: Can't attach it to a transaction?

Comment (nuanyang233) [reply packages/web3-providers/src/types/AbstractAccount.ts:75]: The final `gasCurrency` is passed through `confirmRequest`.

File: packages/plugins/SmartPay/src/SNSAdaptor/components/SmartPayContent.tsx [hunk 4]
```diff
@@ -308,21 +317,20 @@ export const SmartPayContent = memo(() => {
...
-            return prev
-        })
-    }, [contractAccounts])
+    // useEffect(() => {
+    //     if (!contractAccounts?.length) return
+    //     if (account) return
+    //     const contractAccount = first(contractAccounts)
+    //     if (!contractAccount) return
+    //     setAccount(contractAccount.address)
+    // }, [contractAccounts, account])
```
Comment (guanbinrui) [inline packages/plugins/SmartPay/src/SNSAdaptor/components/SmartPayContent.tsx]:
```suggestion
```

File: packages/plugins/SmartPay/src/SNSAdaptor/components/SmartPayContent.tsx [hunk 5]
```diff
@@ -145,61 +153,62 @@ export const SmartPayContent = memo(() => {
...
     const [addAnchorEl, setAddAnchorEl] = useState<HTMLElement | null>(null)
     const [manageAnchorEl, setManageAnchorEl] = useState<HTMLElement | null>(null)
-    const [current, setCurrent] = useState<Wallet>()
 
     const wallets = useWallets()
     const contractAccounts = wallets.filter((x) => x.owner)
 
     // #region Remote Dialog Controller
-    const { openDialog: openApproveMaskDialog } = useRemoteControlledDialog(PluginSmartPayMessages.approveDialogEvent)
+    // const { openDialog: openApproveMaskDialog } = useRemoteControlledDialog(PluginSmartPayMessages.approveDialogEvent)
```
Comment (guanbinrui) [inline packages/plugins/SmartPay/src/SNSAdaptor/components/SmartPayContent.tsx]:
```suggestion
```

File: packages/mask/src/extension/popups/pages/Wallet/ContractInteraction/index.tsx [hunk 6]
```diff
@@ -259,11 +300,23 @@ const ContractInteraction = memo(() => {
     // Wei
     const gasPriceEIP1559 = new BigNumber(maxFeePerGas ?? defaultPrices?.maxFeePerGas ?? 0, 16)
     const gasPricePriorEIP1559 = new BigNumber(gasPrice ?? defaultPrices?.gasPrice ?? 0)
-    const gasFee = (isSupport1559 ? gasPriceEIP1559 : gasPricePriorEIP1559)
-        .multipliedBy(gas ?? 0)
-        .integerValue()
-        .toFixed()
 
+    const gasFee = useMemo(() => {
+        if (!gas) return new BigNumber(0)
```
Comment (guanbinrui) [inline packages/mask/src/extension/popups/pages/Wallet/ContractInteraction/index.tsx]:
```suggestion
        if (!gas) return ZERO
```

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**